src/utils/password.spec.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { randomBytes } from 'crypto';
+
+import { bufferToBase64Url, checkPassword, hashPassword } from './password';
```

Comment:
There should be an additional chrck to make sure we use bcrypt. Currently, if bcrypt broke and return the string itself, we wouldn't notice, would we?

Or to put it more into perspective of test-driven development: If one would like to update the password schema, would they need to touch the tests?

Reply:
I could add
```
expect(hash).not.toEqual(testPassword);
```

Reply:
Then you just verify that thr password is not just returned, but you actually want to verify that it's using a bcrypt hash.

We need an actual hashed result of the password and compare it to the output of the password hash function.

Reply:
> Then you just verify that the password is not just returned, but you actually want to verify that it's using a bcrypt hash.

We added a test that the bcrypt hash looks right (via a regex)

> We need an actual hashed result of the password and compare it to the output of the password hash function.

As bcrypt doesn't compute the same result twice. I'm not sure if we really want that. I guess we would need to mock the PRNG used in bcrypt to achive that.…

Reply:
IMHO it's not our responsibility to test node functions. Or other packages. It should be enough to test if the bcrypt function was called.

Reply:
So.. mock the bcrypt function and check if it was called.

Reply:
iirc jest can check if a function as called without it being necessary to be mocked.